Classes/Utility/AttributeUtility.php

```diff
@@ -41,6 +41,44 @@ public static function findAttribute(int $attributeId, string $selectFields = '*
         );
     }
 
+    /**
+     * @param int $productId
+     * @param string $selectFields
+     * @return array|null
+     */
+    public static function findAllAttributesForProduct(int $productId, string $selectFields = '*'): ?array
+    {
+        $queryBuilder = self::getQueryBuilderForTable(AttributeValueRepository::TABLE_NAME);
+
+        $attributesIds = $queryBuilder
+            ->select('attribute')
+            ->from(AttributeValueRepository::TABLE_NAME)
+            ->where(
+                $queryBuilder->expr()->eq('product', $queryBuilder->createNamedParameter($productId))
+            )
+            ->execute()
+            ->fetchAllAssociativeIndexed();
+
+        $attributesIds = array_keys($attributesIds);
+
+        $queryBuilder = self::getQueryBuilderForTable(AttributeRepository::TABLE_NAME);
+
+        $attributes = $queryBuilder
+            ->select(...GeneralUtility::trimExplode(',', $selectFields, true))
+            ->from(AttributeRepository::TABLE_NAME)
+            ->where(
+                $queryBuilder->expr()->in('uid', $attributesIds)
+            )
+            ->execute()
+            ->fetchAllAssociativeIndexed();
```

Comment:
* Make this into one join statement

(Remember it's $attributeIds, not $attributesIds)